src/lib/market-service/portals/portals.ts

```diff
@@ -0,0 +1,309 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import {
+  arbitrumChainId,
+  ASSET_NAMESPACE,
+  avalancheChainId,
+  baseChainId,
+  bscChainId,
+  ethChainId,
+  fromAssetId,
+  gnosisChainId,
+  optimismChainId,
+  polygonChainId,
+  toAssetId,
+} from '@shapeshiftoss/caip'
+import type {
+  AssetsByIdPartial,
+  FindAllMarketArgs,
+  HistoryData,
+  MarketCapResult,
+  MarketData,
+  MarketDataArgs,
+  PriceHistoryArgs,
+} from '@shapeshiftoss/types'
+import { HistoryTimeframe } from '@shapeshiftoss/types'
+import Axios from 'axios'
+import { setupCache } from 'axios-cache-interceptor'
+import dayjs from 'dayjs'
+import qs from 'qs'
+import { zeroAddress } from 'viem'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { assertUnreachable, getTimeFrameBounds, isToken } from 'lib/utils'
+
+import generatedAssetData from '../../asset-service/service/generatedAssetData.json'
+import type { MarketService } from '../api'
+import { DEFAULT_CACHE_TTL_MS } from '../config'
+import { isValidDate } from '../utils/isValidDate'
+import type { GetTokensResponse, HistoryResponse } from './types'
+
+const calculatePercentChange = (openPrice: string, closePrice: string): number => {
+  const open = bnOrZero(openPrice)
+  const close = bnOrZero(closePrice)
+  if (open.isZero()) return 0
+  return close.minus(open).dividedBy(open).times(100).toNumber()
+}
+
+export const CHAIN_ID_TO_PORTALS_NETWORK: Partial<Record<ChainId, string>> = {
```

Comment:
We already have this in `src/lib/market-service/portals/constants.ts`, yeah?

Reply:
bda92632